subsys/net/ip/net_pkt.c

```diff
@@ -1337,6 +1337,16 @@ int net_pkt_alloc_buffer_with_reserve(struct net_pkt *pkt,
 	return 0;
 }
 
+static bool is_pkt_tx(struct net_pkt *pkt)
+{
+#if defined(CONFIG_NET_CONTEXT_NET_PKT_POOL)
+	if ((pkt->context != NULL) && (get_tx_slab(pkt->context) != NULL)) {
```

Comment:
We call `get_tx_slab()` twice here which seems pointless. I think this is enough
```
	if (pkt->context != NULL) {
		return pkt->slab == get_tx_slab(pkt->context);
	}
```

Reply:
I don't think that'd work, if the slab is not set for the context we should fallback to the common `tx_pkts` slab. With the proposed solution we'd return NULL instead.

Reply:
Indeed, never mind my comment then.